### PR TITLE
update for changing any items to *

### PR DESCRIPTION
**Pull request by hanbollar**
Fixes [#5411](https://github.com/AnalyticalGraphicsInc/cesium/issues/5411)

Updated to match correct changes; however, I'm not sure every * was put in correctly so i erred more on the side of leaving it as Object over changing to * for most items

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Signed CLA is on file.

@hanbollar, thanks for the pull request! Maintainers, we have a signed CLA from @hanbollar, so you can review this at any time.

:warning: I noticed that [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) has not been updated. If this change updates the public API in any way, fixes a bug, or makes any non-trivial update, please add a bullet point to `CHANGES.md` and comment on this pull request so we know it was updated. For more info, see the [Pull Request Guidelines]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines).

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

**Comment by hanbollar** [comment]
@hpinkos @mramato lmk if this matches what you were referencing

**Comment by ggetz** [comment]
@hanbollar The build is erroring when generating the docs, make sure you can run the `generateDocumentation` command locally and everything looks accurate in the built docs.

**Comment by ggetz** [comment]
@hanbollar Give these a try: https://stackoverflow.com/questions/14611995/how-to-specify-an-array-of-objects-as-a-parameter-or-return-value-in-jsdoc

**Comment by hanbollar** [comment]
updated for the fix

File: Source/Core/GoogleEarthEnterpriseTileInformation.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ define([
     /**
      * Creates GoogleEarthEnterpriseTileInformation from an object
      *
-     * @param {Object} info Object to be cloned
+     * @param {*} info Object to be cloned
```
Comment (ggetz) [inline Source/Core/GoogleEarthEnterpriseTileInformation.js]: This one sounds like it still should be of type `Object`. If not, change the doc wording to be more general to all types.

File: Source/Core/IndexDatatype.js [hunk 2]
```diff
@@ -95,7 +95,7 @@ define([
      * or <code>Uint32Array</code> depending on the number of vertices.
      *
      * @param {Number} numberOfVertices Number of vertices that the indices will reference.
-     * @param {*} indicesLengthOrArray Passed through to the typed array constructor.
+     * @param {Object} indicesLengthOrArray Passed through to the typed array constructor.
```
Comment (ggetz) [inline Source/Core/IndexDatatype.js]: This should be a `number` or `Array`, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint32Array

File: Source/Core/Resource.js [hunk 3]
```diff
@@ -159,11 +159,11 @@ define([
     /**
      * This combines a map of query parameters.
      *
-     * @param {Object} q1 The first map of query parameters. Values in this map will take precedence if preserveQueryParameters is false.
-     * @param {Object} q2 The second map of query parameters.
+     * @param {*} q1 The first map of query parameters. Values in this map will take precedence if preserveQueryParameters is false.
```
Comment (ggetz) [inline Source/Core/Resource.js]: Maps imply this should still be an object

File: Source/Core/Resource.js [hunk 4]
```diff
@@ -1570,7 +1570,7 @@ define([
      * using XMLHttpRequest, which means that in order to make requests to another origin,
      * the server must have Cross-Origin Resource Sharing (CORS) headers enabled.
      *
-     * @param {Object} data Data that is posted with the resource.
+     * @param {*} data Data that is posted with the resource.
```
Comment (ggetz) [inline Source/Core/Resource.js]: I _believe_ the data you post must be an `Object` or a `string`.

File: Source/Core/destroyObject.js [hunk 5]
```diff
@@ -23,7 +23,7 @@ define([
      *
      * @exports destroyObject
      *
-     * @param {Object} object The object to destroy.
+     * @param {*} object The object to destroy.
```
Comment (ggetz) [inline Source/Core/destroyObject.js]: The wording implies this should still be an `Object`, I don't think you can destroy primitive types.

File: Source/Core/mergeSort.js [hunk 6]
```diff
@@ -60,7 +60,7 @@ define([
      *
      * @param {Array} array The array to sort.
      * @param {mergeSort~Comparator} comparator The function to use to compare elements in the array.
-     * @param {Object} [userDefinedObject] An object to pass as the third parameter to <code>comparator</code>.
+     * @param {*} [userDefinedObject] An object to pass as the third parameter to <code>comparator</code>.
```
Comment (ggetz) [inline Source/Core/mergeSort.js]: Reword the doc to be more generic than 'object'.

File: Source/Renderer/Context.js [hunk 7]
```diff
@@ -431,7 +431,7 @@ define([
          * be stored globally, except they're tied to a particular context, and to manage
          * their lifetime.
          *
-         * @type {Object}
```
Comment (ggetz) [inline Source/Renderer/Context.js]: I think this is a map, and should remain an object.

File: Source/Scene/DebugCameraPrimitive.js [hunk 8]
```diff
@@ -81,7 +81,7 @@ define([
         /**
          * User-defined object returned when the primitive is picked.
```
Comment (ggetz) [inline Source/Scene/DebugCameraPrimitive.js]: Again, reword the docs here to be more generic than 'object'

File: Source/Scene/DebugModelMatrixPrimitive.js [hunk 9]
```diff
@@ -91,7 +91,7 @@ define([
         /**
          * User-defined object returned when the primitive is picked.
```
Comment (ggetz) [inline Source/Scene/DebugModelMatrixPrimitive.js]: See above

File: Source/Scene/Expression.js [hunk 10]
```diff
@@ -42,7 +42,7 @@ define([
      * @constructor
      *
      * @param {String} [expression] The expression defined using the 3D Tiles Styling language.
-     * @param {Object} [defines] Defines in the style.
+     * @param {*} [defines] Defines in the style.
```
Comment (ggetz) [inline Source/Scene/Expression.js]: This is a map, leave as `Object`.

File: Source/Scene/Label.js [hunk 11]
```diff
@@ -962,7 +962,7 @@ define([
         /**
          * Gets or sets the user-defined object returned when the label is picked.
```
Comment (ggetz) [inline Source/Scene/Label.js]: Update wording.

File: Source/Scene/PointPrimitive.js [hunk 12]
```diff
@@ -424,7 +424,7 @@ define([
         /**
          * Gets or sets the user-defined object returned when the point is picked.
```
Comment (ggetz) [inline Source/Scene/PointPrimitive.js]: Update wording.

File: Source/Scene/Polyline.js [hunk 13]
```diff
@@ -276,7 +276,7 @@ define([
         /**
          * Gets or sets the user-defined object returned when the polyline is picked.
```
Comment (ggetz) [inline Source/Scene/Polyline.js]: wording.

File: Source/ThirdParty/Autolinker.js [hunk 14]
```diff
@@ -527,9 +527,9 @@ Autolinker.Util = {
 	/**
 	 * Assigns (shallow copies) the properties of `src` onto `dest`.
 	 *
-	 * @param {Object} dest The destination object.
-	 * @param {Object} src The source object.
-	 * @return {Object} The destination object (`dest`)
+	 * @param {*} dest The destination object.
```
Comment (ggetz) [inline Source/ThirdParty/Autolinker.js]: Wording.

File: Source/ThirdParty/jsep.js [hunk 15]
```diff
@@ -582,7 +582,7 @@ define(function() {
 	/**
 	 * @method jsep.addLiteral
```
Comment (ggetz) [inline Source/ThirdParty/jsep.js]: Don't update anything in the `ThirdParty` directory- It's third party code, and we don't want to make changes unless necessary.

File: Tools/jsdoc/cesium_template/publish.js [hunk 16]
```diff
@@ -162,7 +162,7 @@ function attachModuleSymbols(doclets, modules) {
 
 /**
  * Create the navigation sidebar.
- * @param {object} members The members that will be used to create the sidebar.
```
Comment (ggetz) [inline Tools/jsdoc/cesium_template/publish.js]: This is also third party code, remove your updates.

File: Apps/Sandcastle/gallery/Custom DataSource.html [hunk 17]
```diff
@@ -252,7 +252,7 @@
 
 /**
  * Loads the provided data, replacing any existing data.
- * @param {Object} data The object to be processed.
+ * @param {*} data The object to be processed.
```
Comment (ggetz) [inline Apps/Sandcastle/gallery/Custom DataSource.html]: Look like data should actually be an array,

**Comment by hanbollar** [comment]
@ggetz updated

File: Source/Core/GoogleEarthEnterpriseMetadata.js [hunk 18]
```diff
@@ -113,7 +113,7 @@ define([
 
         /**
          * Dictionary of provider id to copyright strings.
-         * @type {Object}
+         * @type {*}
```
Comment (hpinkos) [inline Source/Core/GoogleEarthEnterpriseMetadata.js]: Revert this one

File: Source/Core/IonResource.js [hunk 19]
```diff
@@ -28,7 +28,7 @@ define([
      * @constructor
      * @augments Resource
      *
-     * @param {Object} endpoint The result of the Cesium ion asset endpoint service.
+     * @param {*} endpoint The result of the Cesium ion asset endpoint service.
```
Comment (hpinkos) [inline Source/Core/IonResource.js]: Revert this one

File: Source/Core/Resource.js [hunk 20]
```diff
@@ -395,7 +395,7 @@ define([
          * Query parameters appended to the url.
          *
          * @memberof Resource.prototype
-         * @type {Object}
+         * @type {*}
```
Comment (hpinkos) [inline Source/Core/Resource.js]: Revert this one

File: Source/Core/Resource.js (hunk 4 above)
Comment (hpinkos) [reply Source/Core/Resource.js]: Yeah, it should be `Object`

File: Source/Core/Resource.js [hunk 21]
```diff
@@ -1693,7 +1693,7 @@ define([
      * using XMLHttpRequest, which means that in order to make requests to another origin,
      * the server must have Cross-Origin Resource Sharing (CORS) headers enabled.
      *
-     * @param {Object} data Data that is posted with the resource.
+     * @param {*} data Data that is posted with the resource.
```
Comment (hpinkos) [inline Source/Core/Resource.js]: ditto

File: Source/Core/clone.js [hunk 22]
```diff
@@ -9,9 +9,9 @@ define([
      *
      * @exports clone
      *
-     * @param {Object} object The object to clone.
+     * @param {*} object The object to clone.
      * @param {Boolean} [deep=false] If true, all properties will be deep cloned recursively.
-     * @returns {Object} The cloned object.
+     * @returns {*} The cloned object.
```
Comment (hpinkos) [inline Source/Core/clone.js]: Revert these

File: Source/ThirdParty/Autolinker.js [hunk 23]
```diff
@@ -527,9 +527,9 @@ Autolinker.Util = {
 	/**
 	 * Assigns (shallow copies) the properties of `src` onto `dest`.
 	 *
-	 * @param {Object} dest The destination object.
```
Comment (hpinkos) [inline Source/ThirdParty/Autolinker.js]: Revert changes to this file.  We don't need to worry about how third party libraries do their documentation

File: Source/Core/Resource.js (hunk 21 above)
Comment (hanbollar) [reply Source/Core/Resource.js]: updated this one as well

**Comment by hanbollar** [comment]
@hpinkos - reverted the changes per your request

**Comment by hpinkos** [comment]
Thanks @hanbollar!

**Pull request merged**